### PR TITLE
scoverage integration

### PR DESCRIPTION
**Pull request by armanbilge**
Automatically configures scoverage to run in CI.

### CODE REVIEW

**Comment by armanbilge** [comment]
Cool, I think it's working! 😳 except we have no coverage here

**Comment by cquiroz** [comment]
I guess we could write a trivial test to verify? Or rather test it in Lucuma-core

File: build.sbt [hunk 1]
```diff
@@ -15,6 +15,7 @@ lazy val core = project
     addSbtPlugin("de.heikoseeberger"  % "sbt-header"               % "5.6.0"),
     addSbtPlugin("org.portable-scala" % "sbt-scalajs-crossproject" % "1.1.0"),
     addSbtPlugin("org.scala-js"       % "sbt-scalajs"              % "1.8.0"),
+    addSbtPlugin("org.scoverage"      % "sbt-scoverage"            % "1.9.3"),
```
Comment (cquiroz) [inline build.sbt:18]: scoverage used to be problematic in js. Is that still the case?

File: core/src/main/scala/lucuma/sbtplugin/LucumaPlugin.scala [hunk 2]
```diff
@@ -84,6 +86,18 @@ object LucumaPlugin extends AutoPlugin {
...
     )
 
+    lazy val lucumaCoverageSettings = Seq(
+      coverageEnabled := githubIsWorkflowBuild.value, // enable in CI
+      githubWorkflowBuild += WorkflowStep.Sbt(
+        List("coverageReport", "coverageAggregate"),
+        name = Some("Aggregate coverage reports")
+      ),
+      githubWorkflowBuildPostamble += WorkflowStep.Run(
+        List("bash <(curl -s https://codecov.io/bash)"),
```
Comment (cquiroz) [inline core/src/main/scala/lucuma/sbtplugin/LucumaPlugin.scala:96]: Neat

**Review by cquiroz — APPROVED** [review]
LGTM

File: build.sbt (hunk 1 above)
Comment (armanbilge) [reply build.sbt:18]: Seems to work now on JS, I tested it in https://github.com/gemini-hlsw/lucuma-sso/pull/380.

**Pull request merged**